pydantic/fields.py

```diff
@@ -458,6 +461,10 @@ def _type_analysis(self) -> None:  # noqa: C901 (ignore complexity)
         elif issubclass(origin, FrozenSet):
             self.type_ = self.type_.__args__[0]
             self.shape = SHAPE_FROZENSET
+        elif issubclass(origin, OrderedDict):
```

Comment:
please move this down, it will be very rare, so should be checked last or close to last.